### PR TITLE
build: `make tidy` should reports error

### PR DESCRIPTION
**Pull request by SunRunAway**
### What problem does this PR solve?

Problem Summary:

I found that the CI passed even if `go.sum` is not tidied in this [PR](https://github.com/pingcap/tidb/pull/15394/files#diff-f949e2d81c8076ebbf8af38fcbb72c1fR198-R199)
It worked before but changed in https://github.com/pingcap/tidb/pull/13365

### What is changed and how it works?

What's Changed:

check the difference of `go.sum` after `go mod tidy`, and report an error if they are different.

How it Works:
1. mess the go.sum
2. 
```bash
$ make tidy
go mod tidy
./tools/check/check-tidy.sh
Files go.sum and /tmp/go.sum.before differ
make: *** [tidy] Error 1
$ echo $?
2
```

And CI works again, see https://internal.pingcap.net/idc-jenkins/blue/organizations/jenkins/tidb_ghpr_check/detail/tidb_ghpr_check/32682/pipeline/47

### Related changes

- None

### Check List 

Tests 

- Manual test (add detailed scripts or steps below)

### CODE REVIEW

File: tools/check/check-tidy.sh [hunk 1]
```diff
@@ -22,4 +22,6 @@ set -euo pipefail
 # go mod tidy do not support symlink
 cd -P .
 
+cp go.sum /tmp/go.sum.before
 GO111MODULE=on go mod tidy
+diff -q go.sum /tmp/go.sum.before
```
Comment (jackysp) [inline tools/check/check-tidy.sh]: Could we use `git diff --quiet go.sum`?

Comment (SunRunAway) [reply tools/check/check-tidy.sh]: > Yes, with git diff there's no need to copy

If there're some uncommitted and valid changes in `go.sum`, `git diff` may report an unexpected error.

Comment (gregwebs) [reply tools/check/check-tidy.sh]: Why would there be uncommitted changes in go.sum ?

Comment (SunRunAway) [reply tools/check/check-tidy.sh]: Developers don't always `git commit` before running `make check`

Comment (gregwebs) [reply tools/check/check-tidy.sh]: oh, yes. I was thinking of the CI use case.

File: tools/check/check-tidy.sh [hunk 2]
```diff
@@ -22,4 +22,6 @@ set -euo pipefail
 # go mod tidy do not support symlink
 cd -P .
 
+cp go.sum /tmp/go.sum.before
```
Comment (gregwebs) [inline tools/check/check-tidy.sh]: It would be better to add a timestamp to the file name

Comment (SunRunAway) [reply tools/check/check-tidy.sh]: IMHO, it's not a good choice, many temporary files would stay in `/tmp`.

**Pull request merged**